#! /bin/sh /usr/share/dpatch/dpatch-run
## 04_qreal_api_fixes.dpatch by Torsten Marek <shlomme@debian.org>
##
## DP: Exchanges double by qreal in some wrapped methods.

@DPATCH@
diff -urNad python-qt4-4.3~/sip/QtCore/qlist.sip python-qt4-4.3/sip/QtCore/qlist.sip
--- python-qt4-4.3~/sip/QtCore/qlist.sip	2007-09-02 00:35:50.000000000 +0200
+++ python-qt4-4.3/sip/QtCore/qlist.sip	2007-09-02 00:35:53.000000000 +0200
@@ -322,8 +322,8 @@
 %End
 };
 %If (Qt_4_3_0 -)
-// QList<QPair<double, double> > is implemented as a Python list of 2-element tuples.
-%MappedType QList<QPair<double, double> >
+// QList<QPair<float, float> > is implemented as a Python list of 2-element tuples.
+%MappedType QList<QPair<float, float> >
 {
 %TypeHeaderCode
 #include <qlist.h>
@@ -340,7 +340,7 @@
     // Set the list elements.
     for (int i = 0; i < sipCpp->size(); ++i)
     {
-        const QPair<double, double> &p = sipCpp->at(i);
+        const QPair<float, float> &p = sipCpp->at(i);
         PyObject *pobj;
 
         if ((pobj = Py_BuildValue((char *)"dd", p.first, p.second)) == NULL)
@@ -370,16 +370,16 @@
         return 1;
     }
 
-    QList<QPair<double, double> > *ql = new QList<QPair<double, double> >;
+    QList<QPair<float, float> > *ql = new QList<QPair<float, float> >;
  
     for (int i = 0; i < PyList_GET_SIZE(sipPy); ++i)
     {
         PyObject *tup = PyList_GET_ITEM(sipPy, i);
 
-        double first = PyFloat_AsDouble(PyTuple_GET_ITEM(tup, 0));
-        double second = PyFloat_AsDouble(PyTuple_GET_ITEM(tup, 1));
+        float first = PyFloat_AsDouble(PyTuple_GET_ITEM(tup, 0));
+        float second = PyFloat_AsDouble(PyTuple_GET_ITEM(tup, 1));
  
-        ql->append(QPair<double, double>(first, second));
+        ql->append(QPair<float, float>(first, second));
     }
  
     *sipCppPtr = ql;
@@ -389,9 +389,9 @@
 };
 %End
 %If (Qt_4_3_0 -)
-// QList<QPair<double, TYPE> > is implemented as a Python list of 2-element tuples.
-template<double, TYPE>
-%MappedType QList<QPair<double, TYPE> >
+// QList<QPair<float, TYPE> > is implemented as a Python list of 2-element tuples.
+template<float, TYPE>
+%MappedType QList<QPair<float, TYPE> >
 {
 %TypeHeaderCode
 #include <qlist.h>
@@ -408,7 +408,7 @@
     // Set the list elements.
     for (int i = 0; i < sipCpp->size(); ++i)
     {
-        const QPair<double, TYPE> &p = sipCpp->at(i);
+        const QPair<float, TYPE> &p = sipCpp->at(i);
         TYPE *t = new TYPE(p.second);
         PyObject *pobj;
 
@@ -447,12 +447,12 @@
         return 1;
     }
 
-    QList<QPair<double, TYPE> > *ql = new QList<QPair<double, TYPE> >;
+    QList<QPair<float, TYPE> > *ql = new QList<QPair<float, TYPE> >;
  
     for (int i = 0; i < PyList_GET_SIZE(sipPy); ++i)
     {
         PyObject *tup = PyList_GET_ITEM(sipPy, i);
-        double d;
+        float d;
         int state;
 
         d = PyFloat_AsDouble(PyTuple_GET_ITEM(tup, 0));
@@ -466,7 +466,7 @@
             return 0;
         }
 
-        ql->append(QPair<double, TYPE>(d, *t));
+        ql->append(QPair<float, TYPE>(d, *t));
 
         sipReleaseInstance(t, sipClass_TYPE, state);
     }
@@ -524,8 +524,8 @@
     return sipGetState(sipTransferObj);
 %End
 };
-// QList<double> is implemented as a Python list of floats.
-%MappedType QList<double>
+// QList<float> is implemented as a Python list of floats.
+%MappedType QList<float>
 {
 %TypeHeaderCode
 #include <qlist.h>
@@ -561,7 +561,7 @@
     if (sipIsErr == NULL)
         return PyList_Check(sipPy);
 
-    QList<double> *ql = new QList<double>;
+    QList<float> *ql = new QList<float>;
  
     for (int i = 0; i < PyList_GET_SIZE(sipPy); ++i)
         ql->append(PyFloat_AsDouble(PyList_GET_ITEM(sipPy, i)));
diff -urNad python-qt4-4.3~/sip/QtCore/qvector.sip python-qt4-4.3/sip/QtCore/qvector.sip
--- python-qt4-4.3~/sip/QtCore/qvector.sip	2007-09-02 00:35:50.000000000 +0200
+++ python-qt4-4.3/sip/QtCore/qvector.sip	2007-09-02 00:35:53.000000000 +0200
@@ -143,8 +143,8 @@
     return sipGetState(sipTransferObj);
 %End
 };
-// QVector<double> is implemented as a Python list of floats.
-%MappedType QVector<double>
+// QVector<float> is implemented as a Python list of floats.
+%MappedType QVector<float>
 {
 %TypeHeaderCode
 #include <qvector.h>
@@ -180,7 +180,7 @@
     if (sipIsErr == NULL)
         return PyList_Check(sipPy);
 
-    QVector<double> *ql = new QVector<double>;
+    QVector<float> *ql = new QVector<float>;
  
     for (int i = 0; i < PyList_GET_SIZE(sipPy); ++i)
         ql->append(PyFloat_AsDouble(PyList_GET_ITEM(sipPy, i)));
@@ -190,9 +190,9 @@
     return sipGetState(sipTransferObj);
 %End
 };
-// QVector<QPair<double, TYPE> > is implemented as a Python list of 2-element tuples.
+// QVector<QPair<float, TYPE> > is implemented as a Python list of 2-element tuples.
 template<TYPE>
-%MappedType QVector<QPair<double, TYPE> >
+%MappedType QVector<QPair<float, TYPE> >
 {
 %TypeHeaderCode
 #include <qvector.h>
@@ -209,7 +209,7 @@
     // Set the list elements.
     for (int i = 0; i < sipCpp->size(); ++i)
     {
-        const QPair<double, TYPE> &p = sipCpp->at(i);
+        const QPair<float, TYPE> &p = sipCpp->at(i);
         TYPE *pt = new TYPE(p.second);
         PyObject *pobj;
 
@@ -248,14 +248,14 @@
         return 1;
     }
 
-    QVector<QPair<double, TYPE> > *qv = new QVector<QPair<double, TYPE> >;
+    QVector<QPair<float, TYPE> > *qv = new QVector<QPair<float, TYPE> >;
  
     for (int i = 0; i < PyList_GET_SIZE(sipPy); ++i)
     {
         PyObject *tup = PyList_GET_ITEM(sipPy, i);
         int state;
         TYPE *s;
-        QPair<double, TYPE> p;
+        QPair<float, TYPE> p;
 
         p.first = PyFloat_AsDouble(PyTuple_GET_ITEM(tup, 0));
 
diff -urNad python-qt4-4.3~/sip/QtGui/qbrush.sip python-qt4-4.3/sip/QtGui/qbrush.sip
--- python-qt4-4.3~/sip/QtGui/qbrush.sip	2007-09-02 00:35:50.000000000 +0200
+++ python-qt4-4.3/sip/QtGui/qbrush.sip	2007-09-02 00:35:53.000000000 +0200
@@ -138,7 +138,7 @@
 
 QDataStream &operator>>(QDataStream &, QBrush & /Constrained/);
 QDataStream &operator<<(QDataStream &, const QBrush & /Constrained/);
-typedef QVector<QPair<double, QColor> > QGradientStops;
+typedef QVector<QPair<float, QColor> > QGradientStops;
 
 class QGradient
 {
diff -urNad python-qt4-4.3~/sip/QtGui/qgraphicsitemanimation.sip python-qt4-4.3/sip/QtGui/qgraphicsitemanimation.sip
--- python-qt4-4.3~/sip/QtGui/qgraphicsitemanimation.sip	2007-09-02 00:35:50.000000000 +0200
+++ python-qt4-4.3/sip/QtGui/qgraphicsitemanimation.sip	2007-09-02 00:35:53.000000000 +0200
@@ -64,19 +64,19 @@
 
 public:
 %If (Qt_4_3_0 -)
-    QList<QPair<double, QPointF> > posList() const;
+    QList<QPair<float, QPointF> > posList() const;
 %End
 %If (Qt_4_3_0 -)
-    QList<QPair<double, double> > rotationList() const;
+    QList<QPair<float, float> > rotationList() const;
 %End
 %If (Qt_4_3_0 -)
-    QList<QPair<double, QPointF> > translationList() const;
+    QList<QPair<float, QPointF> > translationList() const;
 %End
 %If (Qt_4_3_0 -)
-    QList<QPair<double, QPointF> > scaleList() const;
+    QList<QPair<float, QPointF> > scaleList() const;
 %End
 %If (Qt_4_3_0 -)
-    QList<QPair<double, QPointF> > shearList() const;
+    QList<QPair<float, QPointF> > shearList() const;
 %End
 
 private:
diff -urNad python-qt4-4.3~/sip/QtGui/qpainterpath.sip python-qt4-4.3/sip/QtGui/qpainterpath.sip
--- python-qt4-4.3~/sip/QtGui/qpainterpath.sip	2007-09-02 00:35:50.000000000 +0200
+++ python-qt4-4.3/sip/QtGui/qpainterpath.sip	2007-09-02 00:35:53.000000000 +0200
@@ -194,8 +194,8 @@
     void setCurveThreshold(qreal threshold);
     qreal curveThreshold() const;
     void setDashPattern(Qt::PenStyle);
-    void setDashPattern(const QVector<double> &dashPattern);
-    QVector<double> dashPattern() const;
+    void setDashPattern(const QVector<float> &dashPattern);
+    QVector<float> dashPattern() const;
     QPainterPath createStroke(const QPainterPath &path) const;
 %If (Qt_4_3_0 -)
     void setDashOffset(qreal offset);
diff -urNad python-qt4-4.3~/sip/QtGui/qpen.sip python-qt4-4.3/sip/QtGui/qpen.sip
--- python-qt4-4.3~/sip/QtGui/qpen.sip	2007-09-02 00:35:50.000000000 +0200
+++ python-qt4-4.3/sip/QtGui/qpen.sip	2007-09-02 00:35:53.000000000 +0200
@@ -91,8 +91,8 @@
     void setCapStyle(Qt::PenCapStyle pcs);
     Qt::PenJoinStyle joinStyle() const;
     void setJoinStyle(Qt::PenJoinStyle pcs);
-    QVector<double> dashPattern() const;
-    void setDashPattern(const QVector<double> &pattern);
+    QVector<float> dashPattern() const;
+    void setDashPattern(const QVector<float> &pattern);
     qreal miterLimit() const;
     void setMiterLimit(qreal limit);
     bool operator==(const QPen &p) const;
diff -urNad python-qt4-4.3~/sip/QtGui/qtextoption.sip python-qt4-4.3/sip/QtGui/qtextoption.sip
--- python-qt4-4.3~/sip/QtGui/qtextoption.sip	2007-09-02 00:35:50.000000000 +0200
+++ python-qt4-4.3/sip/QtGui/qtextoption.sip	2007-09-02 00:35:53.000000000 +0200
@@ -62,8 +62,8 @@
     typedef QFlags<QTextOption::Flag> Flags;
     QFlags<QTextOption::Flag> flags() const;
     qreal tabStop() const;
-    void setTabArray(QList<double> tabStops);
-    QList<double> tabArray() const;
+    void setTabArray(QList<float> tabStops);
+    QList<float> tabArray() const;
     void setUseDesignMetrics(bool b);
     bool useDesignMetrics() const;
     void setAlignment(Qt::Alignment aalignment);
